src/helpers/FormValidators.ts

```diff
@@ -0,0 +1,64 @@
+import { parse, isDate } from "date-fns";
+import * as yup from "yup";
+
+const parseDateString = (value: string, originalValue: string) => {
+    const parsedDate = isDate(originalValue)
+        ? originalValue
+        : parse(originalValue, "yyyy-MM-dd", new Date());
+
+    return parsedDate;
+};
+
+const today = new Date()
+
+
+// Dynamically build schema from schema properties
+export const buildSchema = (attributes: any) => {
+    let schemaObject = {};
+
+    for (const prop in attributes) {
+        if (
+            attributes[prop].format === "date" ||
+            attributes[prop].format === "date-time"
+        ) {
+            schemaObject = {
+                ...schemaObject,
+                [prop]: yup.date().transform(parseDateString).max(today).required(),
```

Comment:
@aurelticot,

that's another use case, I guess I will have to remove the `max` date so that this field can be flexible to handle all kinds of data fields in the future?

Reply:
It's a generic validation function, so, theoretically, must adapt to all known and unknown use cases.
We may lack details from the schema standards to perform proper validations.

But, as I said elsewhere, it's a demo repo handling a known scope of use cases. Not a big deal to keep it like you did.